engine/src/flutter/display_list/dl_text.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_DISPLAY_LIST_DL_TEXT_H_
+#define FLUTTER_DISPLAY_LIST_DL_TEXT_H_
+
+#include "flutter/display_list/geometry/dl_geometry_types.h"
+#include "third_party/skia/include/core/SkRefCnt.h"
+
+namespace impeller {
+class TextFrame;
+}
+
+class SkTextBlob;
+
+namespace flutter {
+class DlText {
+ public:
+  virtual DlRect getBounds() const = 0;
+  virtual std::shared_ptr<impeller::TextFrame> getTextFrame() const = 0;
+  virtual sk_sp<SkTextBlob> getTextBlob() const = 0;
```

Comment:
One of the magic things about sk_sp is that you can just use a raw pointer in cases where you are just passing it along (because the ref count is built-in and there is free "shared from this"). So, this could return SkTextBlob*. Indeed, if you look at SkCanvas::drawTextBlob, the real method takes the ptr and the version that takes an sk_sp does a get() on it and passes it to the pointer version...